x-pack/plugins/cloud_integrations/cloud_chat/public/plugin.tsx

```diff
@@ -57,31 +56,64 @@ export class CloudChatPlugin
     this.config = initializerContext.config.get();
   }
 
-  public setup(core: CoreSetup, { cloud, security, cloudChatProvider }: CloudChatSetupDeps) {
-    this.setupChat({ http: core.http, cloud, security, cloudChatProvider }).catch((e) =>
+  public setup(core: CoreSetup, { cloud, security }: CloudChatSetupDeps) {
+    this.setupChat({ http: core.http, cloud, security }).catch((e) =>
       // eslint-disable-next-line no-console
       console.debug(`Error setting up Chat: ${e.toString()}`)
     );
+  }
+
+  public start(core: CoreStart, { cloud, cloudExperiments }: CloudChatStartDeps) {
     const CloudChatContextProvider: FC = ({ children }) => {
       // There's a risk that the request for chat config will take too much time to complete, and the provider
       // will maintain a stale value.  To avoid this, we'll use an Observable.
       const chatConfig = useObservable(this.chatConfig$, undefined);
       return <ServicesProvider chat={chatConfig}>{children}</ServicesProvider>;
     };
-    cloud.registerCloudService(CloudChatContextProvider);
-    cloudChatProvider.registerChatProvider(() => this.Chat);
-  }
+    function ConnectedChat() {
+      const [variant, setVariant] = React.useState<ChatVariantExperiment | null>(null);
+      React.useEffect(() => {
+        if (cloudExperiments) {
+          cloudExperiments
+            .getVariation<ChatVariantExperiment>('cloud-chat.chat-variant', 'header-menu-item')
+            .then((_variant) => {
+              setVariant(_variant === 'floating-bubble' ? 'floating-bubble' : 'header-menu-item');
+            })
+            .catch(() => {
+              setVariant('header-menu-item');
+            });
+        } else {
+          setVariant('header-menu-item');
+        }
+      }, []);
+
+      return (
+        <CloudChatContextProvider>
+          <KibanaThemeProvider theme$={core.theme.theme$}>
```

Comment:
This context is deprecated.  I don't think you even need this, or the `i18nProvider`, if this is indeed an in-band render.

https://github.com/elastic/kibana/blob/main/packages/react/kibana_context/README.mdx

Reply:
Thanks! I'll take look. Not sure I understand what you mean by "in-band" render (this is rendered in a separate mount point)

Reply:
Should I use KibanaRenderContextProvider instead?